src/pages/heeyeolKim/Login/Login.js

```diff
@@ -1,109 +1,105 @@
 import React from 'react';
 
 import { withRouter } from 'react-router-dom';
-//import { Link } from 'react-router-dom';
-
-import '../Login/Login.scss';
 
 import appStore1 from '../../../images/heeyeolKim/Login/appStore1.png';
 import googlePlay2 from '../../../images/heeyeolKim/Login/googlePlay2.png';
 import loginPageImage from '../../../images/heeyeolKim/Login/loginPageImage.png';
 import logoImage from '../../../images/heeyeolKim/Login/logoImage.png'
 
+import '../Login/Login.scss';
+
 class Login extends React.Component {
     constructor() {
         super();
         this.state = {
             idValue: "",
             pwValue: "",
-            bottonOpacity: "", // 버튼 활성화를 위한 state
-            buttonOn: false, // 메인으로 가기 위한 state. input에 요건이 충족하면 opacity가 1이 되고, buttonOn이 true가 되어 메인으로 갈 수 있다.
+            buttonToGo: "buttonOff",
         }
     }
 
-    // handleActivation = () => {
-    //     //const validEmailRegex = RegExp(/^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i);
-    //     if (this.state.idValue.includes("@") && this.state.pwValue.length >= 5) {
+    handleActivation = () => {
+        const { idValue, pwValue } = this.state;
+        let isLoginValid = idValue.includes("@") && pwValue.length >= 5;
+
+        isLoginValid ? this.setState({buttonToGo: "buttonOn"}) : this.setState({buttonToGo: "buttonOff"})
+    }
```

Comment:
```suggestion
    handleActivation = () => {
        const { idValue, pwValue } = this.state;
        const isLoginValid = idValue.includes("@") && pwValue.length >= 5;
        
        // 1차 리팩토링
        this.setState({
            buttonToGo : isLoginValid ? "buttonOn" : "buttonOff"
        })
        
        // 2차 리팩토링
        this.setState({
            buttonToGo : isLoginValid
        })
    }
```
- 삼항 연산자를 value 지정하는 부분에서 바로 적용할 수 있습니다.
- 해당 데이터는 두 가지 경우 밖에 없기 때문에 boolean 데이터 타입을 활용하는 것이 적절해 보입니다.
- isLoginValid 자체가 Boolean 데이터이기 때문에 바로 buttonToGo의 value로 지정해줄 수 있겠네요!

Reply:
2차 리팩토링의 방식으로 코드 수정 했습니다! 조건문을 변수에 담는 것과, boolean 데이터 타입을 활용해 훨씬 깔끔한 코드가 가능한 것 같습니다.